engines/registration_sharing/app/controllers/registration_sharing/smt_to_rmt_controller.rb

```diff
@@ -0,0 +1,101 @@
+require_dependency 'registration_sharing/application_controller'
+
+module RegistrationSharing
+
+  class SmtToRmtException < RuntimeError; end
+
+  class SmtToRmtController < ApplicationController
+    def regsvc
+      unless check_ip
+        render status: :forbidden, plain: 'Forbidden'
+        return
+      end
+
+      xml = Nokogiri::XML(request.body.read)
+
+      if params[:command] == 'shareregistration'
+        return smt_share_registration(xml)
+      elsif params[:command] == 'deltesharedregistration'
+        return smt_delete_registration(xml)
+      else
+        raise SmtToRmtException, 'Command not supported'
+      end
+    rescue SmtToRmtException => e
+      render status: :bad_request, plain: e.to_s
+    end
+
+    protected
+
+    def check_ip
+      allowed_ips = (Settings[:regsharing][:smt_allowed_ips] rescue [] || [])
+      allowed_ips.include?(request.remote_ip)
```

Comment:
This is going to break in GCE. We moved GCE to a different model since their published IP address data is a mess. In GCE there is a script running on the platform that generates firewall rules. This script is run by the Google team and we do not restrict access on our end. We have to account for this case, either by allowing the check to be skipped via config, or by recognizing an entry such as 0.0.0.0/0

Reply:
That's the analogue of `acceptRegistrationSharing` in `/etc/smt.conf` -- I see that it is set on GCE SMTs. This parameter holds IPs of SMTs which RMT will accept shared registrations from. For restricting client IP ranges I've added optional nginx config includes as part of `strict_authentication` plugin.

RMT to RMT sharing uses a shared secret instead (due to reverse-lookups and other considerations from the latest SMT CVE).

Reply:
OK, yes now that you say it it is obvious, must have had something else in my head at the time.